proselint/tools.py

```diff
@@ -365,6 +365,52 @@ def existence_check(text, list, err, msg, ignore_case=True, str=False,
     return errors
 
 
+def _case_sensitive_allowed_word(permitted: set[str], match: re.Match):
+    """Determine if a match object result is in a set of strings."""
+    matched = match.string[match.start():match.end()]
+    return matched in permitted
+
+
+def _case_insensitive_allowed_word(permitted: set[str], match: re.Match):
+    """Determine if a match object result is in a set, ignoring case."""
+    matched = match.string[match.start():match.end()].lower()
+    return matched in permitted
```

Comment:
these should arguably be merged and use an `ignore_case` flag, like the existence check itself does.

Reply:
Hm, not a bad idea, I like it! I'll do that.